### PR TITLE
Add replaced and overridden states for port_group module

### PR DESCRIPTION
**Pull request by mingjunzhang2019**
SUMMARY
This is to add "replaced" and "overridden" states support for sonic_port_group module.

GitHub Issues
N/A

ISSUE TYPE
- Feature Pull Request

COMPONENT NAME
sonic_port_group

OUTPUT
- regression test result: 
[regression-2023-04-06-09-53-35.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11179998/regression-2023-04-06-09-53-35.html.pdf)
- regression test log: 
[regression_test.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11179999/regression_test.log)

ADDITIONAL INFORMATION
- unit test log: 
[unit_test.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11180007/unit_test.log)
- unit test yaml file: 
[port_group.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11180009/port_group.txt)

Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [x] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

How Has This Been Tested?
Tested on real SONIC switch.

### CODE REVIEW

File: plugins/modules/sonic_port_group.py [hunk 1]
```diff
@@ -221,6 +221,100 @@
...
+#3           Ethernet8 - Ethernet11     10G, 25G          25G           10G
+#4           Ethernet12 - Ethernet15    10G, 25G          25G           25G
+#5           Ethernet16 - Ethernet19    10G, 25G          25G           10G
+#6           Ethernet20 - Ethernet23    10G, 25G          25G           25G
+#7           Ethernet24 - Ethernet27    10G, 25G          25G           10G
+#8           Ethernet28 - Ethernet31    10G, 25G          25G           10G
+#9           Ethernet32 - Ethernet35    10G, 25G          25G           10G
+#10          Ethernet36 - Ethernet39    10G, 25G          25G           10G
+#
+- name: Overridden port group speed
```
Comment (ArunSaravananBalachandran) [inline plugins/modules/sonic_port_group.py:291]: Please change to 'Override'.

File: tests/regression/roles/sonic_port_group/defaults/main.yml [hunk 2]
```diff
@@ -47,6 +47,36 @@ tests:
...
+      - id: "{{ pg1 }}"
+        speed: SPEED_25GB
+      - id: "{{ pg2 }}"
+        speed: SPEED_10GB
+      - id: "{{ pg16 }}"
+        speed: SPEED_10GB
+      - id: "{{ pg20 }}"
+        speed: SPEED_10GB
+  - name: test_case_07
+    description: Overridden port group speeds
```
Comment (ArunSaravananBalachandran) [inline tests/regression/roles/sonic_port_group/defaults/main.yml:70]: Please change to 'Override'.

**Review by ArunSaravananBalachandran — CHANGES_REQUESTED** [review]

**Comment by ArunSaravananBalachandran** [comment]
Please add changelog fragment for this PR.

File: plugins/modules/sonic_port_group.py (hunk 1 above)
Comment (mingjunzhang2019) [reply plugins/modules/sonic_port_group.py]: Changed.

File: tests/regression/roles/sonic_port_group/defaults/main.yml (hunk 2 above)
Comment (mingjunzhang2019) [reply tests/regression/roles/sonic_port_group/defaults/main.yml]: Changed.

**Comment by ArunSaravananBalachandran** [comment]
> > 
> 
> changelog fragment for this PR has been added in a general PR for fragments.

Thanks

**Pull request merged**